gentl/src/ffi/mod.rs

```diff
@@ -242,7 +242,7 @@ impl CopyTo for &str {
                 if *dst_size < string_len {
                     return Err(GenTlError::BufferTooSmall);
                 }
-                std::ptr::copy_nonoverlapping(self.as_ptr().cast::<i8>(), dst, self.len());
+                std::ptr::copy_nonoverlapping(self.as_ptr().cast::<i8>(), dst.cast::<i8>(), self.len());
```

Comment:
Right, `libc::c_char` can be an `i8` or a `u8` depending on the platform.

`&str.as_ptr()` always returns a [*const u8](https://doc.rust-lang.org/std/primitive.str.html#method.as_ptr) as far as I'm aware, so it almost makes more sense to cast `dst` to a `u8` pointer and remove the cast on `self.as_ptr()`.

To be honest this whole function feels a bit precarious and should probably be using [`CString`](https://doc.rust-lang.org/std/ffi/struct.CString.html) and [`CStr`](https://doc.rust-lang.org/std/ffi/struct.CStr.html) but I guess that's a bigger conversation.

@strohel what do you think?

Reply:
Yeah, if my understanding is correct that the type doesn't matter as long as it's 8-bit and src and dst matches, then I think `u8` would be more natural.

Alternatively, this should also work I think:
```suggestion
                std::ptr::copy_nonoverlapping(self.as_ptr().cast::<libc::c_char>(), dst, self.len());
```